libc/include/assert.h.def

```diff
@@ -12,4 +12,15 @@
 // This file may be usefully included multiple times to change assert()'s
 // definition based on NDEBUG.
 
+
+#undef assert
+#ifdef NDEBUG
+#define assert(e) (void)0
+#else
+
+_Noreturn void __assert_fail(const char *, const char *, unsigned, const char *) __NOEXCEPT;
```

Comment:
this needs the `ifdef __cplusplus` guards so that it's properly `extern "C"` in C mode